src/common/middlewares/validations.ts

```diff
@@ -0,0 +1,67 @@
+import { BadRequestError } from '@map-colonies/error-types';
+import { polygonPartsEntityNameSchema, VALIDATIONS } from '@map-colonies/mc-model-types';
+import { z, ZodError, type ZodType } from 'zod';
+import type { GetAggregationLayerMetadataHandler } from '../../aggregation/controllers/aggregationController';
+import type { CreatePolygonPartsHandler, UpdatePolygonPartsHandler } from '../../polygonParts/controllers/polygonPartsController';
+import { getEntitiesNames } from '../../polygonParts/DAL/utils';
+import type { EntityName, EntityNames, PolygonPartsPayload } from '../../polygonParts/models/interfaces';
+
+const polygonPartsDBEntityNameSchema: ZodType<EntityName> = z.object({
+  entityName: polygonPartsEntityNameSchema.shape.polygonPartsEntityName,
+  databaseObjectQualifiedName: z.string().regex(new RegExp('^[a-z][a-z0-9_]{0,62}\\.[a-z][a-z0-9_]{0,62}$')),
+});
+
+const partsDBEntityNameSchema: ZodType<EntityName> = z.object({
+  entityName: z.string().regex(new RegExp(VALIDATIONS.polygonPartsEntityName.pattern)),
+  databaseObjectQualifiedName: z.string().regex(new RegExp('^[a-z][a-z0-9_]{0,62}\\.[a-z][a-z0-9_]{0,62}$')),
+});
+
+const entityNamesSchema: ZodType<EntityNames> = z
+  .object({
+    parts: partsDBEntityNameSchema,
+    polygonParts: polygonPartsDBEntityNameSchema,
+  })
+  .strict();
+
+const parsePolygonPartsEntityName = (polygonPartsPayload: PolygonPartsPayload): EntityNames => {
+  const entityNames = getEntitiesNames(polygonPartsPayload);
+  return entityNamesSchema.parse(entityNames);
+};
+
+export const validateGetAggregationLayerMetadata: GetAggregationLayerMetadataHandler = (req, _, next) => {
+  try {
+    polygonPartsEntityNameSchema.shape.polygonPartsEntityName.parse(req.params.polygonPartsEntityName);
```

Comment:
I would recommend check the param object

![image](https://github.com/user-attachments/assets/4e5a6aee-0221-4461-8ab5-e03844532d24)

Reply:
done